### ISSUE TITLE
Lazyloading the srcset

### ISSUE THREAD

**Issue by AlchemyUnited** [scottjehl/picturefill#413]
Hello again.

I did see this other issue but I'm thinking it might be a bit dated at this point. 

https://github.com/scottjehl/picturefill/issues/265

And I did also find this, but it is also even older

http://stackoverflow.com/questions/20507164/is-there-a-way-to-combine-the-lazyload-js-library-with-picturefill

I have a hacky proof of concept where I use a data attribute to store the scrset and then use that to populate the real srcset. I'm presuming I can apply this in a scr= LazyLoad sorta way. But before I dive in, I want to ask, does this sound reasonable (i.e., doable)? Do you think I can LazyLoad the srcset instead of the src? Or might I run into some crazy / stupid browser quirk somewhere along the way and end up wasting my time. 

I also asking because it seems like something like this would be no brainer. Could it get any better than picture / picturefill and lazyload? feels like the ultimate image combo to me. But since I'm not finding it I'm thinking others have tried but that it's not really possible. 

Thanks in advance for your thoughts / help.

**Comment by aFarkas** [comment]
Other tried and [it worked](https://github.com/aFarkas/lazysizes).

**Comment by AlchemyUnited** [comment]
@aFarkas - Good morning.

1) Geez. I should have known. Forgive me for my silliness :)

2) Can I make a suggestion? Or at least explore an alt-idea / implementation with you? 

Basically, what I'm thinking is that it's universally a two part problem.

1) Offset / distance distance from the vert center of the viewport. Note: NOT the top and bottom but the center. Let me explain...

2) So you tag / select the elements you want offset tracked. As they are monitored, you can hang other events off that. Let me explain (and some pseudo code)...

For example, let's say an image is offset .5 * window.height + 500 from the v-center and you know from a data element that at some defined (v-center offset value) point you want to lazyload that image. Great! 

BUT what if you also have a div with some text in it. And at some defined point you want to add / remove a class. And that adjustment triggers a CSS transform. For example, the div / text enters the viewport and I want to apply a class that will cause an animate it in some way. Lets say the test moves from right to left or something like that. Or maybe I want to do something with an image after it lazyloads but hit another offset on the screen. 

The point is, offset tracking has other more universal uses. I'm not sure about actual implementation / coding - yet. This was just an idea I had late last night. It makes sense to me because not only do I need lazyload but I also have other v-offset "event" needs so it makes sense to build the tracker and then use trigger()s (?) to supplement that with specific needs. 

Too ambitious? Too resource heavy? Too much a pain in the arse? Yes, I understand, the possibility can be endless, but that kinda the point. Lazyload, or something else, or something, is simply the result (so to speak) about an element hitting a spot on the page. 

Am I explaining well? What do you think?

**Comment by aFarkas** [comment]
@AlchemyUnited 
I don't really understand what you mean. But can you simply open an issue in the correct repo for this: https://github.com/aFarkas/lazysizes/issues

Additionally in case you want to make some testing before you make this suggestion take the newest version and not the tagged version (I did a lot of improvement there).

**Comment by AlchemyUnited** [comment]
@aFarkas  - Okay thanks. I'll have a look. 

In the mean time, let me try to explain again. 

Lazyload is the results of an event / rule. That is, when the image is here a X...lazyload. Well, why limit the result of that evaluation to lazyload. Why not track / monitor offset and then act based on that? Maybe it's Lazyload. Maybe it's addClass() that kicks off a CSS transform. Or both? Maybe a certain offset tkicks off a Google Analytics event? Regardless of what happens, it all starts with the tracking / knowing the offset.  

It's a two part problem / solution (or could be), not only a single purpose one. Part One (v-offset monitoring) is universal. Then there are "speciality apps" (so to speak) that lean on that foundation (rather than repeat the foundation in each "app"). I want to solve for all offset "events", not just for Lazyload

Maybe that helped?

**Comment by aFarkas** [comment]
@AlchemyUnited 
As I said, we can discuss this here: https://github.com/aFarkas/lazysizes/issues
But I think, I got something for you.
